ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGField.scala

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.postgresql
+
+import java.lang
+import java.time.{Instant, ZoneOffset}
+import java.time.format.DateTimeFormatter
+
+import scala.reflect.ClassTag
+
+/** Type parameters TO and CONVERTED need to comply to the following contract:
+  * - either both of them nullable (AnyRef)
+  * - or neither of them nullable (CONVERTED AnyVal, TO AnyVal or AnyRef which will be never null)
+  * This is important, since absence is automatically signaled with null references (and we still would want to support primitives)
+  */
+private[postgresql] sealed abstract class PGField[FROM, TO, CONVERTED](implicit
```

Comment:
I took me a while to understand the type parameters here. Is the following correct?
- `FROM`: an arbitrary type. The database field value can somehow be extracted from these objects.
- `TO`: a type describing the content of the database field
- `CONVERTED`: the actual SQL type of the database field

For example, a field holding a list of parties would use `TO = List[String]` (because this is how you would store a list of parties in Scala) and `FROM = String` (because what we actually store in the SQL schema is the concatenation of all party strings).

A small description might be useful.

Reply:
yeah unfortunately all these 3 types needed for the purpose, since we want to have one part of the conversion in the type, and another in the definition.

I will beef up the scaladoc in this area